osu.Game/Tests/Visual/TestPlayer.cs

```diff
@@ -18,6 +22,8 @@ namespace osu.Game.Tests.Visual
     /// </summary>
     public class TestPlayer : Player
     {
+        public ISkin Skin { get; set; }
```

Comment:
You're going to need to explain why this is required since I've lost track. Looks *very* flaky.

Reply:
Basically to pass an arbitrary `ISkin` from the test scene to be cached in the `RulesetSkinProvidingContainer` prioritised over the `SkinManager` skins.